VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,989 @@
+by complete enumerations of all impacted versions, such as in the `NVD CPE Match
+feed <https://nvd.nist.gov/vuln/data-feeds#cpeMatch>`_.
+
+Because of this, expressing and resolving a version range is often a complex, or
+error prone task.
+
+In particular the need for common notation for version has emerged based on the
+usage of Package URLs referencing vulnerable package version ranges such as in
+vulnerability databases like `VulnerableCode
+<https://github.com/nexB/vulnerablecode/>`_.
+
+To better understand the problem, here are some of the many notations and
+conventions in use:
+
+- ``semver`` https://semver.org/ is a popular specification to structure version
+  strings, but does not provide a way to express version ranges.
+
+- Rubygems strongly suggest using ``semver`` for version but does not enforce it.
+  As a result some gem use semver while several popular package do not use
+  strict semver. Rubygems use their own notation for version ranges which
+  looks like the ``node-semver`` notation with some subtle differences.
+  See https://guides.rubygems.org/patterns/#semantic-versioning
+
+- ``node-semver`` ranges are used in npm at https://github.com/npm/node-semver#ranges
+  with range semantics that are specific to ``semver`` versions and npm.
+
+- Dart pub versioning scheme is similar to ``node-semver`` and the documentation
+  at https://dart.dev/tools/pub/versioning provides a comprehensive coverage of
+  the topic of versioning. Version resolution uses its own algorithm.
+
+- Python uses its own version and version ranges notation with notable
+  peculiarities on how pre-release and post-release suffixes are used
+  https://www.python.org/dev/peps/pep-0440/
+
+- Debian and Ubuntu use their own notation and are remarkable for their use of
+  ``epochs`` to disambiguate versions.
+  https://www.debian.org/doc/debian-policy/ch-relationships.html
+
+- RPM distros use their own range notation and use epochs like Debian.
+  https://rpm-software-management.github.io/rpm/manual/dependencies.html
+
+- Perl CPAN defines its own version range notation similar to this specification
+  and uses two-segment versions. https://metacpan.org/pod/CPAN::Meta::Spec#Version-Ranges
+
+- Apache Maven and NuGet use similar math intervals notation using brackets
+  https://en.wikipedia.org/wiki/Interval_(mathematics)
+
+  - Apache Maven http://maven.apache.org/enforcer/enforcer-rules/versionRanges.html
+  - NuGet https://docs.microsoft.com/en-us/nuget/concepts/package-versioning#version-ranges
+
+- gradle uses Apache Maven notation with some extensions
+  https://docs.gradle.org/current/userguide/single_versions.html
+
+- Gentoo and Alpine Linux use comparison operators similar to this specification:
+  - Gentoo https://wiki.gentoo.org/wiki/Version_specifier
+  - Alpine linux https://gitlab.alpinelinux.org/alpine/apk-tools/-/blob/master/src/version.c
+
+- Arch Linux https://wiki.archlinux.org/title/PKGBUILD#Dependencies use its
```

Comment:
The man-pages explain the version format a bit better then the PKGBUILD wiki page.

https://man.archlinux.org/man/vercmp.8#DESCRIPTION